src/main/java/io/odpf/firehose/metrics/Metrics.java

```diff
@@ -38,26 +42,67 @@
     public static final String SINK_PUSH_BATCH_SIZE_TOTAL = APPLICATION_PREFIX + SINK_PREFIX + "push_batch_size_total";
 
     // RETRY MEASUREMENT
+    public static final String RETRY_MESSAGES_TOTAL = APPLICATION_PREFIX + RETRY_PREFIX + "messages_total";
     public static final String RETRY_TOTAL = APPLICATION_PREFIX + RETRY_PREFIX + "total";
     public static final String RETRY_SLEEP_TIME_MILLISECONDS = APPLICATION_PREFIX + RETRY_PREFIX + "backoff_sleep_milliseconds";
 
     // DLQ MEASUREMENTS
     public static final String DQL_RETRY_TOTAL = APPLICATION_PREFIX + DLQ_PREFIX + "retry_total";
     public static final String DLQ_MESSAGES_TOTAL = APPLICATION_PREFIX + DLQ_PREFIX + "messages_total";
 
+    // GLOBAL MEASUREMENTS
+    public static final String GLOBAL_MESSAGES_TOTAL = APPLICATION_PREFIX + GLOBAL_PREFIX + "messages_total";
+
     // PIPELINE DURATION MEASUREMENTS
     public static final String PIPELINE_END_LATENCY_MILLISECONDS = APPLICATION_PREFIX + PIPELINE_PREFIX + "end_latency_milliseconds";
     public static final String PIPELINE_EXECUTION_LIFETIME_MILLISECONDS = APPLICATION_PREFIX + PIPELINE_PREFIX + "execution_lifetime_milliseconds";
 
+    // BigQuery SINK MEASUREMENTS
+    public static final String SINK_BIGQUERY_OPERATION_TOTAL = APPLICATION_PREFIX + SINK_PREFIX + BIGQUERY_SINK_PREFIX + "operation_total";
+    public static final String SINK_BIGQUERY_OPERATION_LATENCY_MILLISECONDS = APPLICATION_PREFIX + SINK_PREFIX + BIGQUERY_SINK_PREFIX + "operation_latency_milliseconds";
+
     // ERROR MEASUREMENT
     public static final String ERROR_EVENT = APPLICATION_PREFIX + ERROR_PREFIX + "event";
+    public static final String ERROR_MESSAGES_TOTAL = APPLICATION_PREFIX + ERROR_PREFIX + "messages_total";
 
     // EXECUTION TAGS
     public static final String SUCCESS_TAG = "success=true";
     public static final String FAILURE_TAG = "success=false";
+    public static final String MESSAGE_TYPE_TAG = "type=%s"; // total, success, failure
+    public static final String MESSAGE_SCOPE_TAG = "scope=%s";
+
+    public static final String BIGQUERY_TABLE_TAG = "table=%s";
```

Comment:
we can scope them properly sink specific constants, Core constants

Reply:
yeah. i will put in the different class.